### PR TITLE
[keba] formatting exception fix

### PR DESCRIPTION
**Pull request by tiptick**
for some reason the channels authreq and authon are getting exceptions, so this is fixing the issue. tested on keba P30c.

### CODE REVIEW

**Comment by J-N-K** [comment]
What type of exception do you get? I would assume that this should work out-of-the-box without defining a pattern.

**Comment by tiptick** [comment]
the exception: [ui.internal.items.ItemUIRegistryImpl] - Exception while formatting value 'ON' of item KebaAuthReq with format '%.0f': f != java.lang.String

**Comment by J-N-K** [comment]
`ON` is not a valid value for a `Number` channel. Why is this send to the channel at all?

**Comment by kaikreuzer** [comment]
@tiptick I guess the bug is that those channel types are defined as `Number` and not as `Switch`. According to the docs, those are of type `Switch`.

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 1]
```diff
@@ -221,13 +221,13 @@
 		<item-type>Number</item-type>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]:
```suggestion
		<item-type>Switch</item-type>
```

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 2]
```diff
@@ -221,13 +221,13 @@
...
 		<label>Authentication Required</label>
 		<description>Authentication required</description>
-		<state readOnly="true"></state>
+		<state pattern="%d" readOnly="true"></state>
 	</channel-type>
 	<channel-type id="authon" advanced="true">
 		<item-type>Number</item-type>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]:
```suggestion
		<item-type>Switch</item-type>
```

**Comment by tiptick** [comment]
sorry for long delay, i am testing the suggested changes, will feedback.

**Comment by tiptick** [comment]
testing positive, all good. Please merge.

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 3]
```diff
@@ -218,16 +218,16 @@
 		<state readOnly="false"></state>
 	</channel-type>
 	<channel-type id="authreq" advanced="true">
-		<item-type>Number</item-type>
+		<item-type>Switch</item-type>
 		<label>Authentication Required</label>
 		<description>Authentication required</description>
-		<state readOnly="true"></state>
+		<state pattern="%d" readOnly="true"></state>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: Shouldn't you also revert this change? `%d` isn't a suitable pattern for a Switch.

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 4]
```diff
@@ -218,16 +218,16 @@
...
 	</channel-type>
 	<channel-type id="authon" advanced="true">
-		<item-type>Number</item-type>
+		<item-type>Switch</item-type>
 		<label>Authentication Enabled</label>
 		<description>Authentication enabled</description>
-		<state readOnly="true"></state>
+		<state pattern="%d" readOnly="true"></state>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: Shouldn't you also revert this change? `%d` isn't a suitable pattern for a Switch.
